src/core/structure/sheet.ts

```diff
@@ -803,30 +942,38 @@ export default class Sheet {
   }
 
   private emitSetCellEvent(
-    colKey: ColumnKey,
-    rowKey: RowKey,
-    colPos: number,
-    rowPos: number,
+    columnIndex: number,
+    rowIndex: number,
     cell: Cell | null,
   ) {
     const payload: CoreSetCellPayload = {
-      keyPosition: {
-        rowKey: rowKey,
-        columnKey: colKey,
-      },
-      indexPosition: {
-        column: colPos,
-        row: rowPos,
+      indexInfo: {
+        columnIndex,
+        rowIndex,
       },
       rawValue: cell ? cell.rawValue : "",
       formattedValue: cell ? cell.formattedValue : "",
-      clearCell: cell == null,
-      clearRow: this.rows.get(rowKey) == null,
     };
 
     this.events.emit(new LightsheetEvent(EventType.CORE_SET_CELL, payload));
   }
 
+  private emitSetStyleEvent(
+    columnIndex: number | null,
+    rowIndex: number | null,
+    value: string,
+  ) {
+    const payload: CoreSetStylePayload = {
+      indexInfo: {
+        rowIndex,
+        columnIndex,
+      },
+      value,
+    };
+
+    this.events.emit(new LightsheetEvent(EventType.VIEW_SET_STYLE, payload));
+  }
+
```

Comment:
You can call `generateStyleStringFromMap` in this method instead to reduce duplicate code; it's called every time this method is used. Might be simpler if you make this method take keys instead.